### PR TITLE
feat(authclients): ability to reference a kubeSecret in AuthClient.SecretFromSecret

### PR DESCRIPTION
**Pull request by Dav-14**
- **fix: error description**
- **feat(authclients): ability to create authclient from kubeSecret**

### CODE REVIEW

File: api/formance.com/v1beta1/authclient_types.go [hunk 1]
```diff
@@ -40,14 +41,17 @@ type AuthClientSpec struct {
 	RedirectUris []string `json:"redirectUris,omitempty" yaml:"redirectUris"`
 	// +optional
 	// RedirectUris allow to list allowed post logout redirect uris for the client
-	PostLogoutRedirectUris []string `json:"postLogoutRedirectUris,omitempty" yaml:"PostLogoutRedirectUris"`
+	PostLogoutRedirectUris []string `json:"postLogoutRedirectUris,omitempty" yaml:"postLogoutRedirectUris"`
```
Comment (Dav-14) [inline api/formance.com/v1beta1/authclient_types.go:44]: Fix Auth config.yaml according to https://github.com/formancehq/auth/blob/bce090421ddee5f1ea35669e7d04a4ee150d2051/pkg/client.go#L154

File: api/formance.com/v1beta1/authclient_types.go [hunk 2]
```diff
@@ -76,6 +80,8 @@ type AuthClientStatus struct {
 //+kubebuilder:subresource:status
 //+kubebuilder:resource:scope=Cluster
 //+kubebuilder:printcolumn:name="Stack",type=string,JSONPath=".spec.stack",description="Stack"
+//+kubebuilder:printcolumn:name="Ready",type=string,JSONPath=".status.ready",description="Is ready"
+//+kubebuilder:printcolumn:name="Info",type=string,JSONPath=".status.info",description="Info"
```
Comment (Dav-14) [inline api/formance.com/v1beta1/authclient_types.go:84]: Additional UI info

File: internal/resources/authclients/create.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func Create(ctx core.Context, stack *v1beta1.Stack, owner client.Object, objectN
 			if t.Spec.ID == "" {
 				t.Spec.ID = uuid.NewString()
 			}
-			if t.Spec.Secret == "" {
+			if t.Spec.Secret == "" && t.Spec.SecretFromSecret == nil {
```
Comment (Dav-14) [inline internal/resources/authclients/create.go:23]: Generate a secret only when both option are not specified

File: internal/resources/authclients/init.go [hunk 4]
```diff
@@ -21,15 +21,41 @@ import (
...
 //+kubebuilder:rbac:groups=formance.com,resources=authclients/finalizers,verbs=update
 
 func Reconcile(ctx Context, stack *v1beta1.Stack, authClient *v1beta1.AuthClient) error {
+	if authClient.Spec.SecretFromSecret != nil && authClient.Spec.Secret != "" {
+		return fmt.Errorf("cannot specify signing key using both .spec.SecretFromSecret and .spec.Secret fields")
+	}
+
+	resourceRefName := "client-secret"
+	if authClient.Spec.SecretFromSecret != nil {
+		_, err := resourcereferences.Create(ctx, authClient, resourceRefName, authClient.Spec.SecretFromSecret.Name, &corev1.Secret{}, func(t *v1beta1.ResourceReference) error {
```
Comment (Dav-14) [inline internal/resources/authclients/init.go:44]: Create a resources ref on AuthClient <-> Secret

File: internal/resources/auths/deployment.go [hunk 5]
```diff
@@ -99,6 +116,30 @@ func createDeployment(ctx Context, stack *v1beta1.Stack, auth *v1beta1.Auth, dat
...
+			return acc
+		}, []*v1beta1.AuthClient{})
+
+		if len(clients) > 0 {
+			resourceReferenceList := &v1beta1.ResourceReferenceList{}
+			if err := ctx.GetClient().List(ctx, resourceReferenceList, client.MatchingLabels{
+				authclients.ClientSecretSelector: stack.Name,
+			}); err != nil {
+				return err
+			}
```
Comment (Dav-14) [inline internal/resources/auths/deployment.go:133]: Select appropriate resourceRef by labels

File: api/formance.com/v1beta1/authclient_types.go (hunk 1 above)
Comment (flemzord) [reply api/formance.com/v1beta1/authclient_types.go:44]: Maybe brankin change ? cc @gfyrag

**Pull request merged**